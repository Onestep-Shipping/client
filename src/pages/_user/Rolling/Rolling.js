import './Rolling.css';

import React, { useCallback } from 'react';
import {useHistory, useParams} from 'react-router-dom';

import BookingDisplay from '../../../components/BookingDisplay/BookingDisplay.js';
import Header from '../../../components/Header/Header.js';
import PropTypes from 'prop-types';
import { ROLL_SHIPMENT } from '../../../apollo/mutations/EditShipmentMutation.js';
import ScheduleResultList from '../../../components/ScheduleResultList/ScheduleResultList.js';
import client from '../../../apollo/index.js';

const Rolling = props => {
  const { id } = useParams();
  const history = useHistory();
  const { schedule, schedules } = props.location.state;

  const handleAccept = useCallback(async (newSchedule) => {
    client.mutate({
      mutation: ROLL_SHIPMENT,
      variables: { 
        shipmentId: id,
        newScheduleId: newSchedule._id
      }
    }).then(res => {
      const { rollShipment } = res.data;
      if (rollShipment === "OK") {
        alert("Successfully changed schedule!");
        history.push('/profile');
      } else {
        alert("Some errors have occured. Please try again!");
      }
    })
  }, [history, id]);

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <h1>Current Schedule</h1>
        <BookingDisplay schedule={schedule} fields={5}/>
        <ScheduleResultList 
          action={handleAccept} scheduleList={schedules} remove={schedule}
        />
      </div>
    </div>
  );
};

export default Rolling;

Rolling.propTypes = {
  location: PropTypes.object,
};