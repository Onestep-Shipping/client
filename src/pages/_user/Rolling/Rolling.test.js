import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import Rolling from './Rolling';
import { shallow } from 'enzyme';

let wrapper = shallow(
  <MemoryRouter>  
    <Rolling />
  </MemoryRouter>
);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), 
  useParams: () => ({
    id: '123456',
  }),
}));

describe('Rolling', () => {
  it('should render the Rolling Component correctly', () => {   
    expect(Rolling).toMatchSnapshot();
  });
});