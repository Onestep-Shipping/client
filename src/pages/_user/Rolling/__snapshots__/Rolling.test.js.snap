// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rolling should render the Rolling Component correctly 1`] = `[Function]`;
